### PR TITLE
ALFREDAPI-552 update swagger spec to be oa2 compliant

### PR DESCRIPTION
**Pull request by RVanhuysseXenit**
This has already been done twice without being committed to the main repository. We should make this a permanent change

Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-552

- [X] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [X] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [X] Does the PR comply to REST HTTP result codes policy outlined in the [user guide](https://docs.xenit.eu/alfred-api/user/rest-api/index.html#rest-http-result-codes)?
- [N/A] Is error handling done through a method annotated with `@ExceptionHandler` in the webscript classes?
- [X] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [X] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Alfred API - Changelog
 
+## 5.0.4 (Unreleased)
+
+### Fixed
+* [ALFREDAPI-552](https://xenitsupport.jira.com/browse/ALFREDAPI-552) Make swagger spec oa2 compliant
```
Comment (WimCrols) [inline CHANGELOG.md:6]: oa2 == Open API v2, right? If yes, please put like that in the CHANGELOG.

**Review by WimCrols — DISMISSED** [review]
Hard to give feedback. If this runs in the appropriate tools, then it is good.

**Review by WimCrols — DISMISSED** [review]

File: docs/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    id "base"
+    id "org.hidetake.ssh" version "2.11.2"
+}
+
+def date = new Date()
+def format = new SimpleDateFormat("yyyy-MM-dd")
+
+tasks.register("buildWebsiteScript", Exec) {
+    setGroup "documentation"
+    commandLine 'chmod', '+x', './build-website.sh'
```
Comment (todorinskiz) [inline docs/build.gradle:13]: we can commit +x with Git, no need to execute it every time

File: docs/build.gradle [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    dependsOn buildWebsiteScript
+    doLast {
+        ssh.run {
+            session(remotes.a2hosting) {
+                def remoteHomeDir = execute 'pwd'
+                def today = format.format(date)
+                put from: file("${layout.buildDirectory.get()}/website-alfred-api_${today}.tar.gz"), into: "${remoteHomeDir}/docs.xenit.eu/"
+                execute "mv ${remoteHomeDir}/docs.xenit.eu/alfred-api ${remoteHomeDir}/docs.xenit.eu/alfred-api_${today}_back"
+                execute "mkdir -p ${remoteHomeDir}/docs.xenit.eu/back/"
+                execute "tar vxzf ${remoteHomeDir}/docs.xenit.eu/website-alfred-api_${today}.tar.gz -C ${remoteHomeDir}/docs.xenit.eu/back/"
```
Comment (todorinskiz) [inline docs/build.gradle:43]: the order might be a bit confusing, would be best to first complete the backup, then deploy the new docs?

**Review by todorinskiz — APPROVED** [review]

**Comment by WimCrols** [comment]
Since this is not yet rebased on the 23.x support branch, the builds will not 100% work.

File: docs/build.gradle [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            session(remotes.a2hosting) {
+                def remoteHomeDir = execute 'pwd'
+                def websiteDir = "${remoteHomeDir}/docs.xenit.eu"
+                def today = format.format(date)
+                def websiteTar = "website-alfred-api_${today}.tar.gz"
+                def backupDir = "alfred-api_${today}_back"
+                // Upload
+                put from: file("${layout.buildDirectory.get()}/${websiteTar}"), into: "${websiteDir}/"
+                // Backup old website component to tar
+                execute "mv ${websiteDir}/alfred-api ${remoteHomeDir}/${backupDir}"
```
Comment (WimCrols) [inline docs/build.gradle:45]: Technically the previous tarball is already a backup, but it does not hurt.

**Review by WimCrols — APPROVED** [review]

**Pull request merged**